datafusion/expr/src/function.rs

```diff
@@ -42,7 +42,7 @@ pub type ReturnTypeFunction =
 
 /// Factory that returns an accumulator for the given aggregate, given
 /// its return datatype.
-pub type AccumulatorFunctionImplementation =
+pub type AccumulatorFactoryFunction =
```

Comment:
This was just misleading, so I changed the name

Reply:
Totally agree – this name is much clearer